lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useRef, useContext, useEffect, useState } from 'react';
+import classnames from 'classnames';
+
+import { nxToastPropTypes, NxToastProps } from './types';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import ToastContext from './contexts';
+import { toastTypeMap } from './toastTypeMapping';
+
+const NxToast = (props: NxToastProps) => {
+  const { toastId, className, message, type, ...otherProps } = props,
+      toastClass = toastTypeMap[type].class,
+      toastIcon = toastTypeMap[type].icon,
+      toastIconLabel = toastTypeMap[type].iconLabel,
+      [visible, setVisible] = useState(false),
+      [toastIsActive, setToastIsActive] = useState(false),
```

Comment:
I believe these two can be combined.  See that they always change values together.

Reply:
> I believe these two can be combined. See that they always change values together.

After testing, it appears that both state are identical in the whole lifecycle of the toast, so I removed the [visible] useState here: [fb6bee0](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/fb6bee00dcb86372dab01a40fd84cfa1460b3ed0)